File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    
    stack = deque()
    
    for c in s:
        if c == '(':
            stack.append(c)
        elif c == ')':
            if len(stack) > 0 and stack[-1] == '(':
                stack.pop()
            else:
                return False

    return True if len(stack) == 0 else False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Deque<Character> stack = new ArrayDeque<>();
        
        for(int i = 0; i < s.length(); i++){
            if(!stack.isEmpty() && stack.peekLast() == '(' && s.charAt(i) == ')'){
                stack.removeLast();  
                continue;
            }
            
            stack.addLast(s.charAt(i));
        }
        
        return stack.isEmpty();
    }
}